packages/util-use-case/lib/enhancers/UseCaseWithLogger.mjs

```diff
@@ -39,22 +38,20 @@ export class UseCaseWithLogger extends UseCase {
 
     log.trace('Start use case execution')
 
-    // TODO: UseCaseWithLogger - Measure execution time
-    // - use json payload
-    // - pass params and results to logger,
-    // - handle error scenario (try/catch)
-    // - cover with unit tests
-    // label: tech-dept
-
+    const startTime = new Date().getTime()
+    const payload = {
+      useCase: this.context.name,
+      params,
+    }
     try {
-      const result = await this.context.origin.run(params);
-
+      const result = await this.context.origin.run(params)
       log.info('Finish use case execution')
+      log.debug({...payload, result, runtime: new Date().getTime() - startTime})
 
       return result
-
-    } catch (error){
+    } catch (error) {
       log.error('Issues with running use case')
```

Comment:
check error for subclass of `ApplicationError` and  log  with  `warn` level, otherwise with `log level error` level